frontend/src/metabase/dashboard/grid-utils.ts

```diff
@@ -0,0 +1,94 @@
+import _ from "underscore";
+
+import { DEFAULT_CARD_SIZE } from "metabase/lib/dashboard_grid";
+import { getVisualizationRaw } from "metabase/visualizations";
+import type {
+  BaseDashboardCard,
+  DashboardTabId,
+} from "metabase-types/api/dashboard";
+
+import { generateMobileLayout } from "./components/grid/utils";
+
+export function getLayoutForDashCard(
```

Comment:
These are just copy-and-pasted from the instance methods of DashboardGrid and make into a dependency-free pure function, should be 100% identical as the one in the class.